test_runner/fixtures/metrics.py

```diff
@@ -34,3 +37,27 @@ def parse_metrics(text: str, name: str = ""):
             metrics.metrics[sample.name].append(sample)
 
     return metrics
+
+
+PAGESERVER_PER_TENANT_METRICS = [
```

Comment:
It's a bit sad we need to maintain this list as part of the tests. Whenever we add a new metric, we need to remember to add it here too. The test failure will remind about that, but it's still an extra hurdle.

How about just checking that the tenant has *some* metrics after you create it, and it has *no* metrics after you delete it?

Reply:
What would be the workflow to verify a new metric is present or some metric isn't dropped on refactorings if we don't list explicitly what we should have? Personally I found it easiest to add the list to tests when starting the refactoring so I was able to catch easy mistakes that happened on the way, such as copypasting from existing STORAGE_TIME vs STORAGE_IO_TIME metric, which leads to completely valid code, but isn't correct.